ts/features/bonus/siciliaVola/screens/voucherList/VoucherListScreen.tsx

```diff
@@ -0,0 +1,204 @@
+  isLoading,
+  isReady,
+  isUndefined
+} from "../../../bpd/model/RemoteValue";
+import { svGenerateVoucherStart } from "../../store/actions/voucherGeneration";
+import { useIODispatch } from "../../../../../store/hooks";
+import { confirmButtonProps } from "../../../bonusVacanze/components/buttons/ButtonConfigurations";
+import FooterWithButtons from "../../../../../components/ui/FooterWithButtons";
+import EmptyListImage from "../../../../../../img/bonus/siciliaVola/emptyVoucherList.svg";
+import { InfoScreenComponent } from "../../../../../components/infoScreen/InfoScreenComponent";
+import { LoadingErrorComponent } from "../../../bonusVacanze/components/loadingErrorScreen/LoadingErrorComponent";
+import { possibleVoucherStateSelector } from "../../store/reducers/voucherList/possibleVoucherState";
+
+type Props = ReturnType<typeof mapDispatchToProps> &
+  ReturnType<typeof mapStateToProps>;
+
+const RenderItemBase = (voucher: VoucherPreview): React.ReactElement | null => (
+  <ListItemComponent
+    title={voucher.destination}
+    subTitle={formatDateAsLocal(voucher.departureDate, true, true)}
+    onPress={() => true}
+  />
+);
+
+/**
+ * In order to optimize the rendering of the item, we use the keyId as unique identifier to avoid to redraw the component.
+ * The trx data cannot change while consulting the list and we use this information to avoid a deep comparison
+ */
+export const RenderItem = React.memo(
+  RenderItemBase,
+  (prev: VoucherPreview, curr: VoucherPreview) =>
+    prev.idVoucher !== curr.idVoucher
+);
+
+/**
+ * Loading item, placed in the footer during the loading of the next page
+ * @constructor
+ */
+const FooterLoading = () => (
+  <>
+    <View spacer={true} />
+    <ActivityIndicator
+      color={"black"}
+      accessible={false}
+      importantForAccessibility={"no-hide-descendants"}
+      accessibilityElementsHidden={true}
+      testID={"activityIndicator"}
+    />
+  </>
+);
+
+const EmptyVoucherList = () => {
+  const dispatch = useIODispatch();
+
+  return (
+    <>
+      <InfoScreenComponent
+        image={<EmptyListImage width={104} height={104} />}
+        title={I18n.t("bonus.sv.voucherList.emptyList.title")}
+        body={I18n.t("bonus.sv.voucherList.emptyList.subtitle")}
+      />
+      <FooterWithButtons
+        type={"SingleButton"}
+        leftButton={confirmButtonProps(
+          () => dispatch(svGenerateVoucherStart()),
+          I18n.t("bonus.sv.voucherList.emptyList.cta")
+        )}
+      />
+    </>
+  );
+};
+
+const VoucherListScreen = (props: Props): React.ReactElement => {
+  const { showAnimatedModal, hideModal } = useContext(LightModalContext);
+
+  const vouchers = toArray(props.indexedVouchers);
+
+  useEffect(() => {
+    props.requestVoucherState();
+    props.resetFilter();
+  }, []);
+
+  useEffect(() => {
+    if (isUndefined(props.requiredDataLoaded)) {
+      props.requestVoucherPage(props.filters);
+    }
+  }, [props.filters, props.requiredDataLoaded]);
```

Comment:
If `/beneficiario/ricercaVoucher` fails at the first call, there is no way to reload and make another call. This could be handled automatically tracking the first load and retry with `props.requestVoucherPage(props.filters)` if `isFirstLoading && (undefined || error)`.


https://user-images.githubusercontent.com/26501317/133289840-be2f239f-d296-4e69-9070-2204b8181023.mov

Reply:
Maybe in this case it's better create a specific page where we inform the user and give the possibility to retry manually?
What do you think?

Reply:
What you think about using the already existing component `LoadingErrorComponent` used in `VoucherListScreen`? 
Atm the loading and error condition of this screen is only binded to the `possibleVoucherState`. Maybe could be used to ensure that all the required initial data (`possibleVoucherState` and the first page of `indexedVouchers`) has been loaded. In case of error, when the user tap on `retry` should be handled the loading only of the missing data. What you think, could be work?

Reply:
Yes, make sense! Updated in [14d2594](https://github.com/pagopa/io-app/pull/3361/commits/14d2594995982020f86315e929b9d696b36f4b72)